src/coreclr/gc/gcinterface.dac.h

```diff
@@ -46,9 +46,15 @@ class dac_heap_segment {
 // of a generation and its allocation context.
 class dac_generation {
 public:
-    gc_alloc_context allocation_context;
-    DPTR(dac_heap_segment) start_segment;
-    uint8_t* allocation_start;
+#define ALL_FIELDS
+#define DEFINE_FIELD(field_name, field_type) field_type field_name;
```

Comment:
>DEFINE_FIELD [](http://example.com/codeflow?start=8&length=12)

I see you are defining these macros again here....duplicated from gc.cpp. can you define them all in this header so you don't have to duplicate the definition in gc.cpp?

Reply:
The definition here and the definition in `gc.cpp` are different, they have the same name but different content. Here we are using the macro to define the fields, but there we are using them to generate array values.

Reply:
> they have the same name but different content

doesn't that cause unnecessary confusion? if they are different we could give them different names...

Reply:
That is the only way I can think of to have a single copy of that list to fields to generate three different sets of things. The field declaration, the array index offset, as well as the loading code in the DAC. 

This is based on what @noahfalk suggested in https://github.com/dotnet/runtime/pull/56056#issuecomment-889682100

Reply:
ahh got it. that's fine then.